Doc/whatsnew/3.12.rst

```diff
@@ -622,6 +622,18 @@ New Features
   ``__dict__`` and weakrefs with less bookkeeping,
   using less memory and with faster access.
 
+* API for performing calls using
+  :ref:`the vectorcall protocol <vectorcall>` was added to the
+  :ref:`Limited API <stable>`:
+
+  * :c:func:`PyObject_Vectorcall`
+  * :c:func:`PyObject_VectorcallMethod`
+  * :const:`Py_TPFLAGS_HAVE_VECTORCALL`
+
+  Together with the previous change, this covers both the incoming and outgoing
+  ends of the vector call protocol. (Contributed by Wenzel Jakob in
+  :gh:`98586`.)
```

Comment:
I wonder if it would be better to expand on the previous bullet point here. With this change, we're now mentioning twice that `Py_TPFLAGS_HAVE_VECTORCALL` have been added to the Limited API.

@CAM-Gerlach?

Reply:
Sorry, my bad (and thanks for checking things so thoroughly!)

This should be `PY_VECTORCALL_ARGUMENTS_OFFSET`, whose cross-referencing was fixed as part of this PR.

Reply:
While unrelated to this PR, one bit of nuance that I never understood is why some of Python's macro constants are prefixed `Py_` and others are prefixed `PY_`.

Reply:
Thanks for fixing that! I still wonder if it makes sense to merge those two bullet points. Alternatively, we could leave them like they are, but I don't like the "Together with the previous change" part. It is not clear what "the previous change" is, which is why I think merging them would make sense. Either that, or spell it out: "together with the following APIs ...". Or just drop that sentence, as it is kind of superfluous already.

Reply:
I would probably merge them, and maybe tweak the combined text to be more concise, but that can be done in the review pass prior to the release, and IMO its best to prefer discrete entries when adding them, to avoid merge conflicts and keep things straightforward, and then determine whether they should be combined once editing the final document. What's most important, IMO, is ensuring the entries are clear, explicit and descriptive, such that a non-subject-matter-expert reader or editor (like me) can easily understand the change and its impacts to users without having to try to fill in the blanks themselves. Editorial changes can always be done later, so long as that is in place.

Reply:
> While unrelated to this PR, one bit of nuance that I never understood is why some of Python's macro constants are prefixed Py_ and others are prefixed PY_. thinking

Historical reasons. The [style guide prescribes `Py_`](https://peps.python.org/pep-0007/#naming-conventions) (TIL!), but the general C convention of all-uppercase macros is strong. And it's not likely to be caught in reviews.